cranelift/codegen/src/isa/pulley_shared/lower.isle

```diff
@@ -1369,7 +1369,12 @@
 (rule (lower (scalar_to_vector a @ (value_type $F64)))
   (pulley_vinsertf64 (pulley_vconst128 0) a 0))
 
+;;;; Rules for `shuffle` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(rule (lower (has_type $I8X16 (shuffle a b (u128_from_immediate mask))))
+  (let ((vmask VReg (pulley_vconst128 mask)))
+    (pulley_vshuffle a b vmask)))
```

Comment:
Instead of putting the `mask` into a `VReg` could it be a `u128` immediate to the `vshuffle` instruction itself? That should cut down on opcodes slightly and more closely matches the wasm semantics of the instruction.

Reply:
> could it be a `u128` immediate to the `vshuffle` instruction itself?

I remember not, but let me try again

Reply:
`vshuffle(VReg, VReg, VReg, u128)` would raise this error

```
thread '<unnamed>' panicked at cranelift/codegen/src/isa/pulley_shared/inst/emit.rs:109:9:
encoded inst InstAndKind { inst: Raw { raw: VShuffle { dst: Writable { reg: VReg(p0v) }, src1: VReg(p0v), src2: VReg(p1v), mask: 41362427191743139026751447860679676176 } }, kind: PhantomData<cranelift_codegen::isa::pulley64::Pulley64> } longer than worst-case size: length: 22, Inst::worst_case_size() = 20
```

Reply:
If it's ok, I guess I could raise the inst. size limit.